### DISCUSSION TITLE
can spacy be trained to tell if a name is a seller or buyer?

### DISCUSSION THREAD

**Discussion by DarkSoliditi** [explosion/spaCy#9093] [Help: Model Advice] [answered]
I am wanting to train a model that can recognize buyers and sellers. I was curious if spacy can train ner models that can determine if person, person(s) or company is a Buyer or a Seller before I prepare my training data or if I will have to learn how to use named entity linking to get this behavior.

**Comment by polm** [comment, marked as answer]
This is not an entity linking problem, what you are describing is called "semantic role labeling". We actually just had a thread about this last week in #9034. 

Our typical advice is that this is hard, so you want to avoid it if possible. If you want to do it, there are various approaches, but one is to use generic NER labels that are highly reliable (like PERSON) and then use the dependency parse, checking verbs to establish relations. This approach requires a little understanding of the dependency parse and will have low coverage at first, but it has high precision and is easy to grow and troubleshoot. For more approaches you can see the chapter of Jurafsky & Martin linked in the other thread.

**Reply by DarkSoliditi** [reply]
Awesome thanks so much, I'll try the simpler way then after I get it into production I'll try to scale up to the dependency parser.
